cluster-cm/src/test/java/com/sequenceiq/cloudbreak/cm/polling/task/AbstractClouderaManagerParcelListenerTaskTest.java

```diff
@@ -0,0 +1,103 @@
+package com.sequenceiq.cloudbreak.cm.polling.task;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.cloudera.api.swagger.ParcelResourceApi;
+import com.cloudera.api.swagger.client.ApiClient;
+import com.cloudera.api.swagger.client.ApiException;
+import com.cloudera.api.swagger.model.ApiParcel;
+import com.cloudera.api.swagger.model.ApiParcelState;
+import com.sequenceiq.cloudbreak.cluster.model.ParcelStatus;
+import com.sequenceiq.cloudbreak.cluster.service.ClusterEventService;
+import com.sequenceiq.cloudbreak.cm.ClouderaManagerOperationFailedException;
+import com.sequenceiq.cloudbreak.cm.client.ClouderaManagerApiPojoFactory;
+import com.sequenceiq.cloudbreak.cm.polling.ClouderaManagerCommandPollerObject;
+
+@ExtendWith(MockitoExtension.class)
+abstract class AbstractClouderaManagerParcelListenerTaskTest<T extends AbstractClouderaManagerParcelListenerTask> {
+
+    static final String CLUSTER_NAME = "cluster-name";
+
+    static final String PRODUCT = "product";
+
+    static final String VERSION = "version";
+
+    @Mock
+    ClouderaManagerApiPojoFactory clouderaManagerApiPojoFactory;
+
+    @Mock
+    ClusterEventService clusterEventService;
+
+    @Mock
+    ParcelResourceApi parcelResourceApi;
+
+    T underTest;
+
+    @Mock
+    ClouderaManagerCommandPollerObject pollerObject;
+
+    @Mock
+    ApiClient apiClient;
+
+    @Mock
+    ApiParcel apiParcel;
+
+    @Mock
+    ApiParcelState apiParcelState;
+
+    abstract T setUpUnderTest();
+
+    @BeforeEach
+    void setUp() throws Exception {
+        underTest = setUpUnderTest();
+        lenient().when(pollerObject.getApiClient()).thenReturn(apiClient);
+        lenient().when(clouderaManagerApiPojoFactory.getParcelResourceApi(apiClient)).thenReturn(parcelResourceApi);
+        lenient().when(parcelResourceApi.readParcel(anyString(), anyString(), anyString())).thenReturn(apiParcel);
```

Comment:
minor: `lenient()`-s are not needed here

Reply:
fixed, thanks